android/src/main/java/com/reactnativestripesdk/PaymentLauncherFragment.kt

```diff
@@ -5,83 +5,171 @@ import android.view.LayoutInflater
+  // Used when handling the next action on a payment intent
+  private val handleNextActionClientSecret: String? = null,
 ) : Fragment() {
   private lateinit var paymentLauncher: PaymentLauncher
-  private var clientSecret: String? = null
-  private var promise: Promise? = null
-  private var isPaymentIntent: Boolean = true
+
+  companion object {
+    /**
+     * Helper-constructor used for confirming payment intents
+     */
+    fun forPayment(context: ReactApplicationContext,
+                   stripe: Stripe,
+                   publishableKey: String,
+                   stripeAccountId: String?,
+                   promise: Promise,
+                   paymentIntentClientSecret: String,
+                   confirmPaymentParams: ConfirmPaymentIntentParams): PaymentLauncherFragment {
+      val paymentLauncherFragment = PaymentLauncherFragment(
+        context,
+        stripe,
+        publishableKey,
+        stripeAccountId,
+        promise,
+        paymentIntentClientSecret = paymentIntentClientSecret,
+        confirmPaymentParams = confirmPaymentParams
+      )
+      addFragment(paymentLauncherFragment, context, promise)
+      return paymentLauncherFragment
+    }
+
+    /**
+     * Helper-constructor used for confirming setup intents
+     */
+    fun forSetup(context: ReactApplicationContext,
+                 stripe: Stripe,
+                 publishableKey: String,
+                 stripeAccountId: String?,
+                 promise: Promise,
+                 setupIntentClientSecret: String,
+                 confirmSetupParams: ConfirmSetupIntentParams): PaymentLauncherFragment {
+      val paymentLauncherFragment = PaymentLauncherFragment(
+        context,
+        stripe,
+        publishableKey,
+        stripeAccountId,
+        promise,
+        setupIntentClientSecret = setupIntentClientSecret,
+        confirmSetupParams = confirmSetupParams
+      )
+      addFragment(paymentLauncherFragment, context, promise)
+      return paymentLauncherFragment
+    }
+
+    /**
+     * Helper-constructor used for handling the next action on a payment intent
+     */
+    fun forNextAction(context: ReactApplicationContext,
+                      stripe: Stripe,
+                      publishableKey: String,
+                      stripeAccountId: String?,
+                      promise: Promise,
+                      handleNextActionClientSecret: String): PaymentLauncherFragment {
+      val paymentLauncherFragment = PaymentLauncherFragment(
+        context,
+        stripe,
+        publishableKey,
+        stripeAccountId,
+        promise,
+        handleNextActionClientSecret = handleNextActionClientSecret,
+      )
+      addFragment(paymentLauncherFragment, context, promise)
+      return paymentLauncherFragment
+    }
+
+    private fun addFragment(fragment: PaymentLauncherFragment, context: ReactApplicationContext, promise: Promise) {
+      (context.currentActivity as? AppCompatActivity)?.let {
```

Comment:
The cast should be to `FragmentActivity` not `AppCompatActivity`. The current solution breaks the Flutter integration. See: https://github.com/flutter-stripe/flutter_stripe/issues/771